plugins/modules/identity/keycloak/keycloak_user.py

```diff
@@ -0,0 +1,524 @@
+    access:
+      description: A dict describing the accesses you have to this user based on the credentials used.
+      type: dict
+      returned: always
+      sample:
+        manage: true
+        manageMembership: true
+        view: true
+
+"""
+
+from ansible_collections.community.general.plugins.module_utils.identity.keycloak.keycloak import (
+    KeycloakAPI,
+    camel,
+    keycloak_argument_spec,
+    get_token,
+    KeycloakError,
+)
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """
+    Module execution
+
+    :return:
+    """
+    argument_spec = keycloak_argument_spec()
+
+    meta_args = dict(
+        state=dict(default="present", choices=["present", "absent"]),
+        realm=dict(default="master"),
+        id=dict(type="str"),
+        name=dict(type="str"),
+        attributes=dict(type="dict"),
+        email=dict(type="str"),
+        enabled=dict(type="bool", default=True),
+        first_name=dict(type="str"),
+        last_name=dict(type="str"),
+        required_actions=dict(type="list", elements="str"),
+        credentials=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                type=dict(type="str"),
+                value=dict(type="str", no_log=True),
+                temporary=dict(type="bool", default=False),
+                userLabel=dict(type="str"),
+            ),
+        ),
+        email_verified=dict(type="bool", default=True),
+    )
+
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_one_of=(
+            [
+                ["id", "name"],
+                ["token", "auth_realm", "auth_username", "auth_password"],
+            ]
+        ),
+        required_together=([["auth_realm", "auth_username", "auth_password"]]),
+    )
+
+    result = dict(changed=False, msg="", diff={}, user="")
+
+    # Obtain access token, initialize API
+    try:
+        connection_header = get_token(module.params)
+    except KeycloakError as e:
+        module.fail_json(msg=str(e))
+
+    kc = KeycloakAPI(module, connection_header)
+
+    realm = module.params.get("realm")
+    state = module.params.get("state")
+    uid = module.params.get("id")
+    username = module.params.get("name")
+    attributes = module.params.get("attributes")
+
+    # attributes in Keycloak have their values returned as lists
+    # via the API. attributes is a dict, so we'll transparently convert
+    # the values to lists.
+    if attributes is not None:
+        for key, val in module.params["attributes"].items():
+            module.params["attributes"][key] = (
+                [val] if not isinstance(val, list) else val
+            )
+
+    # Filter and map the parameters names that apply to the user
+    user_params = [
+        x
+        for x in module.params
+        if x not in list(keycloak_argument_spec().keys()) + ["state", "realm"]
```

Comment:
Rather than filtering out what is *not* what you want, I suggest filtering in exactly what is wanted instead.

Reply:
@russoz, I am following the same pattern used in other modules, and I believe its rather better this way because what's wanted will vary from module to module, while what's not wanted is same for all. Here each module imports a keycloak util module which has the functions to perform the actual actions, and it provides the authentication too, when calling the module, for convenience, the auth params are also passed along with module params, but as they are not to be part of resource api's payload, they are excluded. You may check out other modules too and the utils module [plugins/module_utils/identity/keycloak/keycloak.py](https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/identity/keycloak/keycloak.py)